src/SimpleSAML/XHTML/Template.php

```diff
@@ -208,10 +211,26 @@ public function asset(string $asset, ?string $module = null, bool $tag = true):
         $file = new File($file);
 
         $tag = $this->configuration->getVersion();
```

Comment:
I'm not sure what using the version and the md5 of the composer.lock is adding besides complexity.
An md5 hash of the file should work for all cases and CPU-load is _not_ an issue, because md5 is lightning-fast.. I'd reduce the two if-blocks to `$tag = md5_file($file);`

Reply:
Or even better, use `hash_hmac_file` and use the salt as the key-parameter.